rcp/org.testeditor.rcp4.uatests/src/test/java/org/testeditor/rcp4/writetests/AutocompletionProposalTest.tcl

```diff
@@ -0,0 +1,140 @@
+import org.testeditor.rcp4.*
+import org.testeditor.fixture.swt.*
+
+# AutocompletionProposalTest
+
+config TestEditorConfig
+
+// the appropriate aml is added within the setup of this test
+
+* Check that "myApp" ^is proposed ^as component
+
+	Component: ActiveEditor
+	// given
+	- Kill content completely // remove any default generated file content	
+	- Type "# App\n\n* do something\n\nComponent: " into active editor
+	// when
+	- hasMyApp = Has autocompletion proposal "myApp" when entering ""
+	// then
+	- assert hasMyApp
+
+* Given an empty component reference _ content completion will already provide useful proposals _ "with wrong" ^is filtered since no element to satisfy the parameter ^is available
+
+	Component: ActiveEditor
+	// given
+	// when
+	- Use auto completion with "" and select "myApp"
+	- Type "\n" into active editor
+	// then
+	- hasDoSomethingInteraction = Has autocompletion proposal "do something \"count: long\" - test step" when entering ""
+	- hasDoDifferentlyInteraction = Has autocompletion proposal "do differently \"count: long\" - test step" when entering ""
+	- shouldHaveDoWithRight = Has autocompletion proposal "with right <element> - test step" when entering ""
+	- shouldNotHaveDoWithWrong = Has autocompletion proposal "with wrong <element> - test step" when entering ""
+	- assert hasDoSomethingInteraction
+	- assert hasDoDifferentlyInteraction
+	- assert shouldHaveDoWithRight
+	- assert !shouldNotHaveDoWithWrong
+
+* Given a prefix of "- " _ selection of content completion works and text inserted ^does ^not add superfluous dash
+
+	Component: ActiveEditor
+	// given
+	- Type "- " into active editor
+	// when
+	- Use auto completion with "" and select "do something \"count: long\" - test step"
+	- Type "\n" into active editor
+	// then
+	- singleDashedInteraction = Contains editor "\n- do something \"1\""
+	- assert singleDashedInteraction
+
+* Given a prefixed of "- do some" _ assist provides only the ones limited to the ones matching that prefix
+
+	Component: ActiveEditor
+	// given
+	- Type "\n- do some" into active editor
+	// when
+	- hasDoSome = Has autocompletion proposal "do something \"count: long\" - test step" when entering ""
```

Comment:
Some of the TCL tests, like this one, read a bit confusing, because the _when_ actions already read like assertions, only to be followed by the actual assertions in the _then_ part. In this case, maybe it would be more appropriate to have an empty _given_ part, consider the "Type …"-thingy to be the _when_ action, and move everything else (the assertions) into the _then_ part(?)